source/standalone/tutorials/04_sensors/benchmark_cameras.py

```diff
@@ -0,0 +1,621 @@
+# launch omniverse app
+app_launcher = AppLauncher(args_cli)
+simulation_app = app_launcher.app
+
+"""Rest everything follows."""
+
+import numpy as np
+import open3d as o3d
+import random
+import time
+import torch
+from matplotlib import pyplot as plt
+
+import omni.isaac.core.utils.prims as prim_utils
+
+import omni.isaac.lab.sim as sim_utils
+from omni.isaac.lab.assets import RigidObject, RigidObjectCfg
+from omni.isaac.lab.sensors.camera import Camera, CameraCfg, TiledCamera, TiledCameraCfg
+from omni.isaac.lab.sensors.ray_caster import RayCasterCamera, RayCasterCameraCfg, patterns
+from omni.isaac.lab.utils.math import convert_perspective_depth_image_to_orthogonal_depth_image, unproject_depth
+
+
+def create_camera_base(
+    camera_cls: type[Camera | TiledCamera],
+    camera_cfg: type[CameraCfg | TiledCameraCfg],
+    num_cams: int,
+    data_types: list[str],
+    height: int,
+    width: int,
+) -> Camera | TiledCamera | None:
+    """Generalized function to create a camera or tiled camera sensor."""
+    # Determine prim prefix based on the camera class
+    name = camera_cls.__name__
+
+    # Create the necessary prims
+    for idx in range(num_cams):
+        prim_utils.create_prim(f"/World/{name}_{idx:02d}", "Xform")
+
+    # If valid camera settings are provided, create the camera
+    if num_cams > 0 and len(data_types) > 0 and height > 0 and width > 0:
+        cfg = camera_cfg(
+            prim_path=f"/World/{name}_.*/{name}",
+            update_period=0,
+            height=height,
+            width=width,
+            data_types=data_types,
+            spawn=sim_utils.PinholeCameraCfg(
+                focal_length=24, focus_distance=400.0, horizontal_aperture=20.955, clipping_range=(0.1, 1e4)
+            ),
+        )
+        return camera_cls(cfg=cfg)
+    else:
+        return None
+
+
+def create_tiled_cameras(
+    num_cams: int = 2, data_types: list[str] = ["rgb", "depth"], height: int = 100, width: int = 120
+) -> TiledCamera | None:
+    """Defines the tiled camera sensor to add to the scene."""
+    return create_camera_base(TiledCamera, TiledCameraCfg, num_cams, data_types, height, width)
+
+
+def create_cameras(
+    num_cams: int = 2, data_types: list[str] = ["rgb", "distance_to_image_plane"], height: int = 100, width: int = 120
+) -> Camera | None:
+    """Defines the camera sensor to add to the scene."""
+    return create_camera_base(Camera, CameraCfg, num_cams, data_types, height, width)
+
+
+def create_ray_caster_cameras(
+    num_cams: int = 2,
+    data_types: list[str] = ["distance_to_image_plane"],
+    mesh_prim_paths: list[str] = ["/World/ground"],
+    height: int = 100,
+    width: int = 120,
+) -> RayCasterCamera | None:
+    for idx in range(num_cams):
+        prim_utils.create_prim(f"/World/RayCasterCamera_{idx:02d}/RayCaster", "Xform")
+
+    if num_cams > 0 and len(data_types) > 0 and height > 0 and width > 0:
```

Comment:
I wonder if it makes more sense to handle this outside of the method? e.g. only call this method if we have more > 0 num_cams